core/src/main/scala/org/typelevel/paiges/Chunk.scala

```diff
@@ -7,25 +7,31 @@ private[paiges] object Chunk {
   /**
    * Given a width and Doc find the Iterator
    * of Chunks.
+   *
+   * A `trim` document is one where all lines consisting entirely of indentation are
+   * represented by the empty string.  Note that this does *not* in general include lines consisting
+   * solely of whitespace.  E.g. Doc.text(" ") will always insert a space, regardless
+   * of whether the document is trimmed or not.
    */
-  def best(w: Int, d: Doc): Iterator[String] = {
+  def best(w: Int, d: Doc, trim: Boolean): Iterator[String] = {
 
     val nonNegW = w max 0
 
     sealed abstract class ChunkStream
     object ChunkStream {
       case object Empty extends ChunkStream
-      case class Item(str: String, position: Int, stack: List[(Int, Doc)], isBreak: Boolean) extends ChunkStream {
+      case class Item(str: String, position: Int, cache: ChunkStream, stack: List[(Int, Doc)], isBreak: Boolean) extends ChunkStream {
+        def isLine: Boolean = str == null
         def stringChunk: String = if (isBreak) lineToStr(position) else str
-        private[this] var next: ChunkStream = null
+        private[this] var next: ChunkStream = cache
         def step: ChunkStream = {
-          // do a cheap local computation.
-          // lazy val is thread-safe, but more expensive
-          // since everything is immutable here, this is
-          // safe
-          val res = next
-          if (res != null) res
-          else {
+          if (cache != null) {
```

Comment:
Oof, good catch.  Fixed.  

This seems like a performance bug in the non-trim case since next is never consulted and `loop` is called every time `step` is requested.   Are the benchmarks supposed to run automatically?  I looked in the Travis logs and didn't see the results.  I can run them locally but don't know what to compare them to.  It would be very nice to see this version break a benchmark test.